### PR TITLE
perf(fronend): 前端类型调整 #127

### PR DESCRIPTION
**Pull request by jinquantianxia**

### CODE REVIEW

**Comment by tencentblueking-adm** [comment]
[CLA assistant check](https://bk-cla.bktencent.com:443/TencentBlueKing/blueking-dbm?pullRequest=303) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://bk-cla.bktencent.com:443/TencentBlueKing/blueking-dbm?pullRequest=303) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bk-cla.bktencent.com:443/check/TencentBlueKing/blueking-dbm?pullRequest=303) it.</sub>

File: dbm-ui/frontend/src/views/mission/pages/index.vue [hunk 1]
```diff
@@ -104,7 +104,7 @@
   });
   const isSearching = computed(() => {
     const { daterange, searchValues } = state.filter;
-    return daterange.filter(item => item).length > 0 || searchValues.length > 0;
+    return (daterange.length > 0) || searchValues.length > 0;
```
Comment (Daryl0819) [inline dbm-ui/frontend/src/views/mission/pages/index.vue:107]: daterange清空后是['', ''].length > 0 是 true，所以要加filter

**Review by Daryl0819 — CHANGES_REQUESTED** [review]

File: dbm-ui/frontend/src/services/ticket.tsx [hunk 2]
```diff
@@ -174,10 +172,7 @@ export const getTicketHostNodes = (params: TicketNodesParams): Promise<HostNode[
 /**
   * 查询访问源列表
   */
-export const getHostInAuthorize = function <T> (params: T)
-: Promise<{ hosts: HostNode[], ip_whitelist: {ip: string}[] }> {
-  return http.get(`/apis/mysql/bizs/${params.bk_biz_id}/permission/authorize/get_host_in_authorize/`, params);
-};
+export const getHostInAuthorize = (bkBizId: string) => http.get<{ hosts: HostNode[], ip_whitelist: {ip: string}[] }>(`/apis/mysql/bizs/${bkBizId}/permission/authorize/get_host_in_authorize/`);
```
[the remark this answers is not in the document]
Comment (jinquantianxia) [reply dbm-ui/frontend/src/services/ticket.tsx:175]: 我改成对T进行限定

File: dbm-ui/frontend/src/views/event-center/pages/DBHASwitchEvents.vue [hunk 3]
```diff
@@ -285,11 +287,10 @@
           }
 
           return {
-            ...item,
             cost_time: costTime,
             result_info: resultInfo,
           };
-        });
+        }) as TableItem[];
```
[the remark this answers is not in the document]
Comment (jinquantianxia) [reply dbm-ui/frontend/src/views/event-center/pages/DBHASwitchEvents.vue:293]: 已恢复

File: dbm-ui/frontend/src/views/mission/pages/Details.vue [hunk 4]
```diff
@@ -320,18 +320,24 @@
...
   });
-  const getStatusTheme = (isTag = false) => {
+
+  // tag状态
+  const tagStatusTheme = computed(() => {
     const value = baseInfo.value.status;
-    if (isTag && value === 'RUNNING') return 'info';
+    if (value === 'RUNNING') return 'info';
+    return 'danger';
+  });
```
[the remark this answers is not in the document]
Comment (jinquantianxia) [reply dbm-ui/frontend/src/views/mission/pages/Details.vue:329]: 之前逻辑可能没理清，已恢复

File: dbm-ui/frontend/src/views/mission/pages/index.vue (hunk 1 above)
Comment (jinquantianxia) [reply dbm-ui/frontend/src/views/mission/pages/index.vue:107]: 已恢复

**Review by Daryl0819 — APPROVED** [review]

**Pull request merged**